eclipse-collections-code-generator/src/main/resources/impl/bag/mutable/primitiveHashBag.stg

```diff
@@ -881,6 +885,14 @@ public class <name>HashBag
             this.canRemove = false;
         }
     }
+<if(primitive.specializedStream)>
+
+    @Override
+    public Spliterator.Of<name> spliterator()
+    {
+        return Spliterators.spliterator(this.items.keySet().toArray(), Spliterator.SIZED);
```

Comment:
This is a very costly implementation of a spliterator and I also believe it is incorrect for a Bag. By using only the keySet, items in the bag will be missed, as the counts could be more than 1.

Reply:
Without custom spliterators, this PR will not be worthwhile. It would be providing an API that would never be desirable to use because of the performance cost.

Reply:
hi @donraab  - i've been scratching my head of how this spliterator will look like and here are some thoughts/observations:

1) In all the implementations i've come across you rely on some type of an array - the HashMap spliterator relies on access to the Nodes[]. The primitive hashmaps (intint for example) that back the bag have keyValues[] structure that provides the storage backing but there's no easy 
2) Java Hashmap also comes with KeySpliterator and ValueSpliterator but I don't think these by themselves will be useful to us in this case because we care about both keys + vals. I was thinking we can possibly get away with doing the spliterator on the key but that's not entirely accurate because i might have 100s of elements in the bag that represent the same thing (100s of 1s for example).
3) I'm wondering if we can do the spliterator in the primitive hashmap, then from the hashbag we will delegate to it?

Happy to explore any other venues you might have in mind too.

Reply:
Hi @Desislav-Petrov, I think maybe we just close this PR for now and wait for someone to request this feature for an actual use case. I think the asymmetry for primitive bags having primitive Stream support will be ok for now.